### PR TITLE
dr-autosync: add metrics

### LINKED ISSUE #6975: metric: add add grafana panel for dr-autosync
## Enhancement Task
It would be convenient  if we add some Grafana panel for  such as dr-autosync status and sync_recover process.

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?

Issue Number: Close #6975

### What is changed and how does it work?

add some dr-autosync related metrics

image

numbers are used to distinguish different states, the map is
```
1: sync
2: async_wait
3: async
4: sync_recover
```

```commit-message
```

### Check List

Tests
- Manual test (add detailed scripts or steps below)

sync state
```
# HELP pd_replication_dr_recover_progress Progress of sync_recover process
# TYPE pd_replication_dr_recover_progress gauge
pd_replication_dr_recover_progress 0
# HELP pd_replication_dr_recovered_region Number of recovered regions
# TYPE pd_replication_dr_recovered_region gauge
pd_replication_dr_recovered_region 0
# HELP pd_replication_dr_state State of DR
# TYPE pd_replication_dr_state gauge
pd_replication_dr_state 1
# HELP pd_replication_dr_state_id State ID of DR
# TYPE pd_replication_dr_state_id gauge
pd_replication_dr_state_id 10
# HELP pd_replication_dr_tick_total Counter of background state check count
# TYPE pd_replication_dr_tick_total counter
pd_replication_dr_tick_total 64
```

async wait
```
# HELP pd_replication_dr_recover_progress Progress of sync_recover process
# TYPE pd_replication_dr_recover_progress gauge
pd_replication_dr_recover_progress 0
# HELP pd_replication_dr_recovered_region Number of recovered regions
# TYPE pd_replication_dr_recovered_region gauge
pd_replication_dr_recovered_region 0
# HELP pd_replication_dr_state State of DR
# TYPE pd_replication_dr_state gauge
pd_replication_dr_state 2
# HELP pd_replication_dr_state_id State ID of DR
# TYPE pd_replication_dr_state_id gauge
pd_replication_dr_state_id 221
# HELP pd_replication_dr_tick_total Counter of background state check count
# TYPE pd_replication_dr_tick_total counter
pd_replication_dr_tick_total 311
```

async
```
# HELP pd_replication_dr_recover_progress Progress of sync_recover process
# TYPE pd_replication_dr_recover_progress gauge
pd_replication_dr_recover_progress 0
# HELP pd_replication_dr_recovered_region Number of recovered regions
# TYPE pd_replication_dr_recovered_region gauge
pd_replication_dr_recovered_region 0
# HELP pd_replication_dr_state State of DR
# TYPE pd_replication_dr_state gauge
pd_replication_dr_state 3
# HELP pd_replication_dr_state_id State ID of DR
# TYPE pd_replication_dr_state_id gauge
pd_replication_dr_state_id 222
# HELP pd_replication_dr_tick_total Counter of background state check count
# TYPE pd_replication_dr_tick_total counter
pd_replication_dr_tick_total 385
```

sync recover
```
# HELP pd_replication_dr_recover_progress Progress of sync_recover process
# TYPE pd_replication_dr_recover_progress gauge
pd_replication_dr_recover_progress 0.24528302252292633
# HELP pd_replication_dr_recovered_region Number of recovered regions
# TYPE pd_replication_dr_recovered_region gauge
pd_replication_dr_recovered_region 0
# HELP pd_replication_dr_state State of DR
# TYPE pd_replication_dr_state gauge
pd_replication_dr_state 4
# HELP pd_replication_dr_state_id State ID of DR
# TYPE pd_replication_dr_state_id gauge
pd_replication_dr_state_id 223
# HELP pd_replication_dr_tick_total Counter of background state check count
# TYPE pd_replication_dr_tick_total counter
pd_replication_dr_tick_total 393
```

sync
```
# HELP pd_replication_dr_recover_progress Progress of sync_recover process
# TYPE pd_replication_dr_recover_progress gauge
pd_replication_dr_recover_progress 1
# HELP pd_replication_dr_recovered_region Number of recovered regions
# TYPE pd_replication_dr_recovered_region gauge
pd_replication_dr_recovered_region 53
# HELP pd_replication_dr_state State of DR
# TYPE pd_replication_dr_state gauge
pd_replication_dr_state 1
# HELP pd_replication_dr_state_id State ID of DR
# TYPE pd_replication_dr_state_id gauge
pd_replication_dr_state_id 224
# HELP pd_replication_dr_tick_total Counter of background state check count
# TYPE pd_replication_dr_tick_total counter
pd_replication_dr_tick_total 614
```

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: pkg/replication/metrics.go [hunk 1]
```diff
@@ -33,3 +57,11 @@ var (
 			Help:      "Progress of sync_recover process",
 		})
 )
+
+func init() {
+	prometheus.MustRegister(drTickCounter)
```
Comment (CabinfeverB) [inline pkg/replication/metrics.go:62]: Is it included in grafana?

File: pkg/replication/metrics.go [hunk 2]
```diff
@@ -33,3 +57,11 @@ var (
...
+	prometheus.MustRegister(drStateGauge)
+	prometheus.MustRegister(drStateIDGauge)
+	prometheus.MustRegister(drRecoveredRegionGauge)
+	prometheus.MustRegister(drRecoverProgressGauge)
```
Comment (CabinfeverB) [inline pkg/replication/metrics.go:66]: ditto

**Review by lhy1024 — APPROVED** [review]

File: pkg/replication/metrics.go (hunk 1 above)
Comment (disksing) [reply pkg/replication/metrics.go:62]: no, wait for me to add one

**Comment by disksing** [comment]
@CabinfeverB added tick rate panel.

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**